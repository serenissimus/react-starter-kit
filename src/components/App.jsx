import 'normalize.css/normalize.css';
import style from 'styles/App.less';
import alive from 'images/alive.jpg';

import PropTypes from 'prop-types';
import React from 'react';

const AppComponent = ({ title }) => {
    return (
        <React.Fragment>
            <h1 className={style.title}>{title}</h1>
            <img src={alive} alt='alive' />
        </React.Fragment>
    );
};

AppComponent.propTypes = {
    title: PropTypes.string.isRequired,
};

AppComponent.defaultProps = {};

export default AppComponent;
