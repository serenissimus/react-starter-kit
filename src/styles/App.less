@import './MaterialColors.less';

.title {
    color: @red-a700;
}
