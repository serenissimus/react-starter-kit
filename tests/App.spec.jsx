import React from 'react';
import renderer from 'react-test-renderer';
import App from 'components/App';

describe('App', () => {
    it('should correct render', () => {
        const component = renderer.create(<App />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should correct render with props', () => {
        const props = {
            title: 'Title',
            logo: 'logo',
        };

        const component = renderer.create(<App {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
