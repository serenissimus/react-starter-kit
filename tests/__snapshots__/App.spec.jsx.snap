// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should correct render 1`] = `
Array [
  <h1
    className="title"
  >
    It's alive!
  </h1>,
  <img
    alt="logo"
    src="test-file-stub"
  />,
]
`;

exports[`App should correct render with props 1`] = `
Array [
  <h1
    className="title"
  >
    Title
  </h1>,
  <img
    alt="logo"
    src="logo"
  />,
]
`;
